ui/app/components/ui/popover/popover.component.js

```diff
@@ -4,12 +4,27 @@ import PropTypes from 'prop-types'
 import classnames from 'classnames'
 import { I18nContext } from '../../../contexts/i18n'
 
-const Popover = ({ title, subtitle, children, footer, footerClassName, onBack, onClose }) => {
+const Popover = ({
+  title,
+  subtitle = '',
+  children,
+  footer,
+  footerClassName,
+  onBack,
+  onClose,
+  className,
+  showTooltip,
+  CustomBackground,
+}) => {
   const t = useContext(I18nContext)
   return (
     <div className="popover-container">
-      <div className="popover-bg" onClick={onClose} />
-      <section className="popover-wrap">
+      { CustomBackground
+        ? <CustomBackground onClose={onClose} />
+        : <div className="popover-bg" onClick={onClose} />
+      }
+      <section className={classnames('popover-wrap', className)}>
+        { showTooltip ? <div className="popover-tooltip" /> : null}
```

Comment:
Speaking of which - it's unfortunate we're using absolute positioning here, rather than dynamically pointing at the thing we want to point at. That's how our tooltips work for example. This will need to be adjusted each time the styles change for the menu bar. It's also not accommodating of translations of different lengths (and neither are the styles for the actual status indicator, I'm just noticing :disappointed: )

We could use a tooltip here instead of our `Popover` component, though that still wouldn't provide an obvious way to position this "hole" in the backdrop :thinking:

Reply:
@Gudahtt My latest commit is a demonstration of how we could position this dynamically. Obviously some code clean up to do, but I'm interested to hear your opinion on this approach taken, and whether it is fundamentally unsound in any way.

ade401d5d664e6fa1c87ece518abfb81d825b93f

Reply:
screenshot:

![Screenshot from 2020-04-09 11-01-01](https://user-images.githubusercontent.com/7499938/78900496-7ad5e800-7a51-11ea-8006-118adf7418fc.png)

Reply:
That is basically how it should be done (use a `ref` to measure the DOM after mount), though that example doesn't take into account the possibility of updates after mounting (e.g. screen resizing).

This problem is already solved for us with the tooltip library though. Mostly I was hoping we could reuse the same library - not re-implement all of this from scratch.

Reply:
except, we have to re-implement some form of it to handle the "hole" in the background right?

Reply:
Yeah, quite possibly. We could cheat our way around it using z-index, but I sure hate suggesting that 😛

Reply:
That could work... not sure if we can cheat our way around dynamic sizing (width, height) though

Reply:
Oh, I guess we could if we don't make a "hole" at all, and instead just style the component using borders/backgrounds to make it look like it is part of the menu-bar, and use z-index to place it above the background :grimacing:

Reply:
I mean, that's not terrible... and less error prone that trying to roll our own dynamic sizer that also handles window resizing

Reply:
Yeah... this seems like the sort of exceptional case that `z-index` might be useful for 😬 

I thought of this because of @whymarrh 's button PR - now that it's a button, we probably want to give it this same oval shape anyway so that it can act like a button (i.e. have reasonable hover/focus states, drop shadow or something when pressed, etc.). So if the underlying component has this oval shape.... we could take advantage of that here.

Reply:
I've opted for my original solution for now, as integrating with react-tippy was becoming cumbersome.

Created a ticket for us to improve this in the future https://github.com/MetaMask/metamask-extension/issues/8332